src/server/pfs/server/driver.go

```diff
@@ -476,17 +513,43 @@ func (d *driver) FinishCommit(ctx context.Context, commit *pfs.Commit) error {
 			if err := tree.DeleteFile(filePath); err != nil {
 				// Deleting a non-existent file in an open commit should
 				// be a no-op
-				if !(hashtree.Code(err) == hashtree.PathNotFound) {
+				if hashtree.Code(err) != hashtree.PathNotFound {
 					return err
 				}
 			}
 		} else {
-			var objectHash string
-			var size int64
-			fmt.Sscanf(string(kv.Value), "%d %s", &size, &objectHash)
-			if err := tree.PutFile(filePath, []*pfs.Object{{Hash: objectHash}}, size); err != nil {
+			records := &PutFileRecords{}
+			if err := proto.Unmarshal(kv.Value, records); err != nil {
 				return err
 			}
+			if !records.Split {
+				if len(records.Records) != 1 {
+					return fmt.Errorf("unexpect %d length PutFileRecord (this is likely a bug)", len(records.Records))
+				}
+				if err := tree.PutFile(filePath, []*pfs.Object{{Hash: records.Records[0].ObjectHash}}, records.Records[0].SizeBytes); err != nil {
+					return err
+				}
+			} else {
+				nodes, err := tree.List(filePath)
+				if err != nil && hashtree.Code(err) != hashtree.PathNotFound {
+					return err
+				}
+				var indexOffset int64
+				if len(nodes) > 0 {
+					indexOffset, err = strconv.ParseInt(path.Base(nodes[len(nodes)-1].Name), 16, 64)
```

Comment:
Should we make these magic numbers into constants?  Wasn't sure what they were at first glance.